packages/perseus-linter/src/__tests__/rule.test.ts

```diff
@@ -52,19 +52,17 @@ the previous heading was level ${previousHeading.level}`;
         },
     ];
 
-    let rules: Array<never> | Array<Rule | any> = [];
-
     function parseTree() {
         return PureMarkdown.parse(markdown);
     }
 
     it("makeRules() factory method", () => {
-        rules = ruleDescriptions.map((r) => Rule.makeRule(r));
+        const rules: Rule[] = ruleDescriptions.map((r) => Rule.makeRule(r));
         expect(rules).toHaveLength(ruleDescriptions.length);
-        rules.forEach((r) => expect(r instanceof Rule).toBeTruthy());
```

Comment:
I wonder if this was intended to verify some sort of edge case. (just me being paranoid)

Reply:
It's duplicating a check that the type system should be doing for us... unless the types are wrong? But it seems pretty clear that `Rule.makeRule` always returns a `Rule`:

```ts
    static makeRule(options: MakeRuleOptions): Rule {
        return new Rule(
            options.name,
            options.severity,
            options.selector ? Selector.parse(options.selector) : null,
            Rule.makePattern(options.pattern),
            options.lint || options.message,
            options.applies,
        );
    }
```

Reply:
...and, now that I'm looking at it again, I see that `expect(rules).toHaveLength(ruleDescriptions.length);` is basically testing `Array.map`. So this whole test should probably be deleted.